demos/coreMQTT/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1916 @@
+        xTaskNotifyWait( 0, mqttexampleUNSUBSCRIBE_COMPLETE_BIT, &ulNotification, mqttexampleDEMO_TICKS_TO_WAIT );
+    }
+
+    LogInfo( ( "Operation wait complete.\n" ) );
+
+    /* Create command to stop command loop. */
+    LogInfo( ( "Beginning command queue termination." ) );
+    prvCreateCommand( TERMINATE, NULL, NULL, &xCommand );
+    xCommandAdded = prvAddCommandToQueue( &xCommand );
+    /* Ensure command was added to queue. */
+    configASSERT( xCommandAdded == pdTRUE );
+
+    /* Notify main task this task can be deleted. */
+    xTaskNotify( xMainTask, mqttexampleSUBSCRIBE_TASK_COMPLETE_BIT, eSetBits );
+
+    /* Delete this task. */
+    LogInfo( ( "Deleting Subscriber task." ) );
+    vTaskDelete( NULL );
+}
+
+/*-----------------------------------------------------------*/
+
+int RunCoreMqttConnectionSharingDemo( bool awsIotMqttMode,
+                                      const char * pIdentifier,
+                                      void * pNetworkServerInfo,
+                                      void * pNetworkCredentialInfo,
+                                      const void * pNetworkInterface )
+{
+    NetworkContext_t xNetworkContext = { 0 };
+    BaseType_t xNetworkStatus = pdFAIL;
+    BaseType_t xResult = pdFALSE;
+    uint32_t ulNotification = 0;
+    Command_t xCommand;
+    MQTTStatus_t xMQTTStatus;
+    int ret = EXIT_SUCCESS;
+
+    ( void ) awsIotMqttMode;
+    ( void ) pIdentifier;
+    ( void ) pNetworkServerInfo;
+    ( void ) pNetworkCredentialInfo;
+    ( void ) pNetworkInterface;
+
+    ulGlobalEntryTimeMs = prvGetTimeMs();
+
+    xMainTask = xTaskGetCurrentTaskHandle();
+
+    /* Create command queue for processing MQTT commands. */
+    xCommandQueue = xQueueCreate( mqttexampleCOMMAND_QUEUE_SIZE, sizeof( Command_t ) );
+    /* Create response queues for each task. */
+    xSubscriberResponseQueue = xQueueCreate( mqttexamplePUBLISH_QUEUE_SIZE, sizeof( PublishElement_t ) );
+    /* Publish task doesn't receive anything in this demo, so it doesn't need a large queue. */
+    xPublisherResponseQueue = xQueueCreate( 1, sizeof( PublishElement_t ) );
+
+    /* In this demo, send publishes on non-subscribed topics to this queue.
+     * Note that this value is not meant to be changed after `prvCommandLoop` has
+     * been called, since access to this variable is not protected by thread
+     * synchronization primitives. */
+    xDefaultResponseQueue = xPublisherResponseQueue;
+
+    /* Connect to the broker. We connect here with the "clean session" flag set
+     * to true in order to clear any prior state in the broker. We will disconnect
+     * and later form a persistent session, so that it may be resumed if the
+     * network suddenly disconnects. */
+    LogInfo( ( "Creating a TCP connection to %s.\r\n", democonfigMQTT_BROKER_ENDPOINT ) );
+    xNetworkStatus = prvConnectNetwork( &xNetworkContext );
+
+    ret = EXIT_FAILURE;
+
+    if( xNetworkStatus == pdPASS )
```

Comment:
I went to office hours and we have decided to keep the loop but limit it by democonfigMQTT_MAX_DEMO_COUNT

Reply:
Is that a requirement? I think the demo's more straightforward without creating/deleting tasks in a loop, but I added the loop back anyway.

Reply:
if the loop exists in FreeRTOS/FreeRTOS then we want to try to keep a semblance of it. Thank you